src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java

```diff
@@ -9,24 +12,76 @@
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import javax.transaction.Transactional;
+import java.util.Optional;
 
 @Service
 @AllArgsConstructor
 public class CustomerServiceImpl implements CustomerService {
 
     private final CustomerRepository customerRepository;
     private final CustomerMapper customerMapper;
+    private final UserServiceImpl userService;
 
     @Override
-    public Page<CustomerDto> getCustomers(final Pageable pageable, final Boolean status) {
+    public Page<CustomerDto> findAllByActive(final Pageable pageable, final Boolean status) {
         return customerRepository.findByStatus(pageable, status).map(customerMapper::mapToCustomerView);
     }
 
     @Override
-    public CustomerDto getCustomer(final Long customerId) {
+    public CustomerDto findById(final Long customerId) {
         return customerRepository.findById(customerId)
                 .map(customerMapper::mapToCustomerView)
                 .orElseThrow(() -> new ResourceNotFoundException("Customer with id=" + customerId + " doesn't exist"));
     }
 
+    @Override
+    public CustomerDto save(CustomerDto customerDto) {
+        Customer customer = Optional.ofNullable(customerDto.getId())
+                .map(item -> {
+                    final Customer existing = customerRepository
+                            .findById(customerDto.getId())
+                            .orElseThrow();
+                    customerMapper.map(customerDto, existing);
+                    return existing;
+                })
+                .orElseGet(() -> customerMapper.map(customerDto));
+
+        final Customer saved = customerRepository.save(customer);
+        userService.save(createAdmin(customerDto));
+        return customerMapper.map(saved);
+    }
+
+    @Transactional
+    public void delete(final Long id) {
+        customerRepository.delete(id);
+    }
+
+    @Override
+    public void activate(Long id) {
+
+    }
+
+    @Transactional
+    public void switchStatus(final Long id) {
+        final Customer existing = customerRepository
+                .findById(id)
+                .orElseThrow();
+        if (existing.isStatus()) {
+            customerRepository.deactivate(id);
```

Comment:
we discussed yesterday that we don't need such a split, we can use only one repo method